Extensions/Today/TodayViewController.swift

```diff
@@ -171,26 +171,13 @@ class TodayViewController: UIViewController, NCWidgetProviding {
         } else {
             self.openCopiedLinkButton.hidden = true
         }
-
-        self.view.setNeedsLayout()
     }
 
     func widgetMarginInsetsForProposedMarginInsets(defaultMarginInsets: UIEdgeInsets) -> UIEdgeInsets {
         TodayUX.defaultWidgetTextMargin = defaultMarginInsets.left
         return UIEdgeInsetsMake(0, 0, TodayUX.verticalWidgetMargin, 0)
     }
 
-    func widgetPerformUpdateWithCompletionHandler(completionHandler: ((NCUpdateResult) -> Void)) {
-        // Perform any setup necessary in order to update the view.
-        dispatch_async(dispatch_get_main_queue()) {
-            // updates need to be made on the main thread
-            self.updateCopiedLink()
-            // and we need to call the completion handler in every branch.
-            completionHandler(NCUpdateResult.NewData)
-        }
-        completionHandler(NCUpdateResult.NewData)
-    }
-
```

Comment:
Is `updateCopiedLink` ever called now?

I'm not sure how this works now, given that that we don't seem to be checking the clipboard. If that's wrong, could you comment how the whole thing works now?